### PR TITLE
Quiet show instances

### PR DESCRIPTION
**Pull request by coot**
Deriving 'Show' instances via 'Quiet', which removes accessors.

### CODE REVIEW

**Comment by coot** [comment]
I am not sure how `SlotNo`, `EpochNo`, `EpochSize`, `BlockNo` `Show` instances could affect the `Hash` tests in `test-crypt` tests.  These tests fail for me also locally on the `master` branch.

**Comment by coot** [comment]
How's possible that [test failed[(https://hydra.iohk.io/build/3536839/nixlog/1) without the last commit.  Is everything fine with CI in this repo?

File: cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs [hunk 1]
```diff
@@ -187,7 +187,7 @@ instance Show (Hash h a) where
 
 instance HashAlgorithm h => IsString (Hash h a) where
   fromString str =
-    case hashFromBytesAsHex (BSC.pack str) of
+    case hashFromBytesAsHex (BSC.pack $ read str) of
```
Comment (newhoggy) [inline cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs:190]: I suspect this is not what we want:

```
λ> "a4612527d3b6018c5290118c48e62fc1" :: Hash MD5 [Char]
"*** Exception: Prelude.read: no parse
λ> "\"a4612527d3b6018c5290118c48e62fc1\"" :: Hash MD5 [Char]
"a4612527d3b6018c5290118c48e62fc1"
it :: Hash MD5 [Char]
(0.00 secs, 1,053,712 bytes)
```

**Comment by newhoggy** [comment]
Please rebase.

File: cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs (hunk 1 above)
Comment (coot) [reply cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs:190]: Right; it's better to fix the `Show` instance.

**Comment by coot** [comment]
@newhoggy do you know why incompatiblitiy of `Show` and `IsString` for `Hash` slipped through CI?

**Comment by coot** [comment]
bors merge

**Comment by newhoggy** [comment]
Related PR https://github.com/input-output-hk/cardano-base/pull/151

**Pull request merged**